bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/AbstractDynamoDBItem.java

```diff
@@ -302,17 +302,17 @@ public static DynamoDBItem<?> fromStateNew(Item item, ZonedDateTime time, @Nulla
         } else if (item instanceof NumberItem) {
             return new DynamoDBBigDecimalItem(name, convert(state, DecimalType.class).toBigDecimal(), time, expireDays);
         } else if (item instanceof PlayerItem) {
-            if (state instanceof PlayPauseType) {
-                switch ((PlayPauseType) state) {
+            if (state instanceof PlayPauseType type) {
+                switch (type) {
                     case PLAY:
                         return new DynamoDBBigDecimalItem(name, PLAY_BIGDECIMAL, time, expireDays);
                     case PAUSE:
                         return new DynamoDBBigDecimalItem(name, PAUSE_BIGDECIMAL, time, expireDays);
                     default:
                         throw new IllegalArgumentException("Unexpected enum with PlayPauseType: " + state.toString());
                 }
-            } else if (state instanceof RewindFastforwardType) {
-                switch ((RewindFastforwardType) state) {
+            } else if (state instanceof RewindFastforwardType type) {
```

Comment:
I see it here and elsewhere. The generated instanceof patterns do not create a meaningfull/logical variable name. 
like 
… instanceof RewindFastforwardType type

Not wrong, but consider more specific bar names.

Reply:
Completely agreed....